Fix some spell check issues

kinow opened the pull request:
A few spell check issues in Javadoc and one in the HTML guide.

Bruno

iorixxx commented:
*TokenizerImpl.java files have corresponding *.jflex files in the source tree. Do you mind fixing typos in *.jflex files? I think the ant target `ant jflex` in `lucene/analysis/common/` is used to generate *.java files from *.jflex files.

kinow commented:
Sure @iorixxx 

Let's see if I didn't miss anything. So first I found the *TokenizerImpl.java files with `git diff --name-only HEAD~1 | grep TokenizerImpl`

```
lucene/analysis/common/src/java/org/apache/lucene/analysis/standard/ClassicTokenizerImpl.java
lucene/analysis/common/src/java/org/apache/lucene/analysis/standard/UAX29URLEmailTokenizerImpl.java
lucene/analysis/common/src/java/org/apache/lucene/analysis/wikipedia/WikipediaTokenizerImpl.java
lucene/core/src/java/org/apache/lucene/analysis/standard/StandardTokenizerImpl.java
```

Then use the file from GitHub master branch.

```
git diff --name-only HEAD~1 | grep TokenizerImpl | xargs -I{} git checkout origin/master {}
git add .
git commit -m "Will be rebased..."
git rebase -i HEAD~2
# mark with f the "Will be rebased commit"
```

And now `git diff --name-only HEAD~1 | grep TokenizerImpl` gives me an empty result.

Confirm I have the jflex files:

```
$git diff --name-only origin/fix-spell-check | xargs -I{} basename {} .java | xargs -I{} find . -name "{}*"
./lucene/analysis/common/src/java/org/apache/lucene/analysis/standard/ClassicTokenizerImpl.jflex
./lucene/analysis/common/src/java/org/apache/lucene/analysis/standard/ClassicTokenizerImpl.java
./lucene/analysis/common/src/java/org/apache/lucene/analysis/standard/UAX29URLEmailTokenizerImpl.java
./lucene/analysis/common/src/java/org/apache/lucene/analysis/standard/UAX29URLEmailTokenizerImpl.jflex
./lucene/analysis/common/src/java/org/apache/lucene/analysis/wikipedia/WikipediaTokenizerImpl.jflex
./lucene/analysis/common/src/java/org/apache/lucene/analysis/wikipedia/WikipediaTokenizerImpl.java
./lucene/core/src/java/org/apache/lucene/analysis/standard/StandardTokenizerImpl.jflex
./lucene/core/src/java/org/apache/lucene/analysis/standard/StandardTokenizerImpl.java
```

Then get the diff (`git diff --patch origin/fix-spell-check `).... and trying to apply the Java changes to jflex files now... except that... there's no "**occured**" on these jflex files...

After a bit of searching here and there, and reading a bit about jflex (had never heard about it, good to learn something new), looks like the typo is actually in jflex's code.

https://github.com/jflex-de/jflex/pull/218

Going to push-force the commit that removes the *TokenizerImpl.java files.

Does it make sense? Did I miss something?

Cheers
Bruno

tflobbe commented on lucene/core/src/java/org/apache/lucene/search/SearcherLifetimeManager.java:
```diff
@@ -238,7 +238,7 @@ public boolean doPrune(double ageSec, IndexSearcher searcher) {
    *  oldest IndexSearcher) order.
    * 
-   *  <p><b>NOTE</b>: you must peridiocally call this, ideally
+   *  <p><b>NOTE</b>: you must periodiocally call this, ideally
```
This should be `periodically`, right?

kinow replied:
+1 will try rebasing it later today/this week

tflobbe commented on lucene/core/src/java/org/apache/lucene/store/IndexInput.java:
```diff
@@ -98,7 +98,7 @@ public IndexInput clone() {
   /**
    * Creates a slice of this index input, with the given description, offset, and length. 
-   * The slice is seeked to the beginning.
+   * The slice is sought to the beginning.
```
I’m wondering if this is intentional to refer to the `seek` method

kinow replied:
Oh, probably sought! Will fix it too.

kinow commented:
@tflobbe rebased (~2500 commits behind :eyes: ). Added the other typos. Squashed commits, and push-force'd. Should be ready to review again. Thanks!
